apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -36,6 +33,86 @@ function prettyCourseName(ltiClaim) {
   }
 }
 
+async function courseInstancesAllowedToLink({
+  course_id,
+  user_id,
+  authn_user_id,
+  is_administrator,
+  authn_is_administrator,
+}: {
+  course_id: string;
+  user_id: string;
+  authn_user_id: string;
+  is_administrator: boolean;
+  authn_is_administrator: boolean;
+}) {
+  const course_instances = await selectCourseInstancesWithStaffAccess({
+    course_id,
+    user_id,
+    authn_user_id,
+    is_administrator,
+    authn_is_administrator,
+  });
+
+  return course_instances.filter((ci) => ci.has_course_instance_permission_edit);
+}
+
+async function coursesAllowedToLink({
+  user_id,
+  is_administrator,
+}: {
+  user_id: string;
+  is_administrator: boolean;
+}) {
+  const courses = await selectCoursesWithEditAccess({
+    user_id,
+    is_administrator,
+  });
+
+  return courses.filter((c) => c.permissions_course.has_course_permission_edit);
+}
+
+router.get(
+  '/course_instances',
+  asyncHandler(async (req, res) => {
+    const courses = await coursesAllowedToLink({
+      user_id: res.locals.authn_user.user_id,
+      is_administrator: res.locals.is_administrator,
+    });
+
+    const course = courses.find((c) => c.id === req.query.unsafe_course_id);
+    if (!course) {
+      throw new HttpStatusError(403, 'Access denied');
+    }
+
+    const course_instances = await courseInstancesAllowedToLink({
+      course_id: course.id,
+      user_id: res.locals.authn_user.user_id,
+      authn_user_id: res.locals.authn_user.user_id,
+      is_administrator: res.locals.is_administrator,
+      authn_is_administrator: res.locals.authn_is_administrator,
+    });
+
+    let options: HtmlSafeString;
+
+    if (course_instances.length === 0) {
+      options = html`<option disabled selected value="">
+        No course instances found where you have student data editor permissions.
+      </option>`;
+    } else {
+      options = course_instances.reduce(
+        (output, ci) => {
+          return html`${output}
+            <option value="${ci.id}">${ci.short_name}: ${ci.long_name}</option>`;
+        },
+        html``,
+      );
```

Comment:
Nit, this is more conventional in our codebase (you'll need to import `joinHtml` from `@prairielearn/html` and maybe reformat this): 

```suggestion
      options = joinHtml(course_instances.map(
        (ci) => {
          return html`<option value="${ci.id}">${ci.short_name}: ${ci.long_name}</option>`;
        },
      ));
```

Reply:
I agree that's more readable. Updated in 633032758